### PR TITLE
Fix notify disassocitaion

### PR DESCRIPTION
**Pull request by takayoshi-k**
## Summary
 After recover disassociation, usrsock daemon must destroy all sockets.
 Add notification to prompt the daemon to do so.

## Impact
Only for gs2200m driver

## Testing
Check if tcpblaster on spresense:wifi work well and if it received an error when disassociation happened.

### CODE REVIEW

File: drivers/wireless/gs2200m.c [hunk 1]
```diff
@@ -2108,6 +2152,35 @@ static enum pkt_type_e gs2200m_activate_wrx(FAR struct gs2200m_dev_s *dev,
...
+ * Name: gs2200m_syncloss
+ * NOTE: See 5.1.4 Sync loss interval
+ ****************************************************************************/
+
+static enum pkt_type_e gs2200m_syncloss(FAR struct gs2200m_dev_s *dev,
+                                            int val)
+{
+  char cmd[30];
+
+  printf("Sync interval = %d x 100ms\n", val);
```
Comment (masayuki2009) [inline drivers/wireless/gs2200m.c:2179]: @takayoshi-k 
printf() should not be used in driver code.
Please replace it with wlinfo().

**Review by masayuki2009 — CHANGES_REQUESTED** [review]

File: drivers/wireless/gs2200m.c (hunk 1 above)
Comment (takayoshi-k) [reply drivers/wireless/gs2200m.c:2179]: @masayuki2009 
Thank you for reviewing.
I had forgotten remove it. I did it and force pushed.
Please check it again.

**Comment by masayuki2009** [comment]
@takayoshi-k 
I noticed that the stress test with spresense:wifi_smp is unstable.
Let me check what is happening with this PR.

File: drivers/wireless/gs2200m.c [hunk 2]
```diff
@@ -2152,6 +2152,35 @@ static enum pkt_type_e gs2200m_activate_wrx(FAR struct gs2200m_dev_s *dev,
   return gs2200m_send_cmd2(dev, cmd);
 }
 
+/****************************************************************************
+ * Name: gs2200m_powersave_wrx
+ * NOTE: See 9.1.1 Active Radio Receive
+ ****************************************************************************/
+
+static enum pkt_type_e gs2200m_powersave_wrx(FAR struct gs2200m_dev_s *dev,
+                                            uint8_t on)
```
Comment (xiaoxiang781216) [inline drivers/wireless/gs2200m.c:2161]: add space after uint8_t

Comment (takayoshi-k) [reply drivers/wireless/gs2200m.c]: @xiaoxiang781216 
Thank you for reviewing.
I add it.
Please check it.

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by masayuki2009** [comment]
>@takayoshi-k
>I noticed that the stress test with spresense:wifi_smp is unstable.
>Let me check what is happening with this PR.

@takayoshi-k 
I did the stress test (nxplayer + telnet) with the first commit only. (i.e. did not apply the apps's PR)
However, it seems unstable.

**Comment by masayuki2009** [comment]
@takayoshi-k 
>I did the stress test (nxplayer + telnet) with the first commit only. (i.e. did not apply the apps's PR)
>However, it seems unstable (i.e. deadlock).

Regarding the first commit, I think it's a little complicated.
I think we don't need `_notif_q_push_wo_enable()`. 
Also, I think we don't need `_enable_cid32()` but just modify `_enable_cid()` to support `uint32_t *cid_bits`.
In this case, we need to modify `uint16_t valid_cid_bits`, `uint16_t aip_cid_bits` as well.

What do you think?

**Comment by takayoshi-k** [comment]
@masayuki2009 
Thank you for your suggestion.
Yes. It can be just expand 16 bits to 17 (atually 32) bits.
I modified it and force pushed.
Please check it again.

**Comment by masayuki2009** [comment]
>I modified it and force pushed.

@takayoshi-k 
Hmm, the deadlock still happens both spresense:wifi_smp and spresenes:wifi (non SMP).
Let me investigate the code later.

**Comment by masayuki2009** [comment]
@takayoshi-k

>Hmm, the deadlock still happens both spresense:wifi_smp and spresenes:wifi (non SMP).
>Let me investigate the code later.

I found that the following change works for both spresense:wifi_smp and spresense:wifi.
Actually, it works with the app's PR. However, I'm still not sure why we need such a change. 

```
diff --git a/drivers/wireless/gs2200m.c b/drivers/wireless/gs2200m.c
index 4ec9d5c9d1..fa2b1d6b76 100644
--- a/drivers/wireless/gs2200m.c
+++ b/drivers/wireless/gs2200m.c
@@ -76,7 +76,7 @@
 
 #define MAX_PKT_LEN  1500
 #define MAX_PAYLOAD  (MAX_PKT_LEN - BULK_CMD_HDR_SIZE_WITH_GUARD)
-#define MAX_NOTIF_Q  17 /* for 16 sockets and disassociation event */
+#define MAX_NOTIF_Q  18 /* for 16 sockets + disassociation event + dummy */
 
 #define WR_REQ       0x01
 #define RD_REQ       0x02
```

**Comment by takayoshi-k** [comment]
@masayuki2009
Thank you for your investigation. 
I also think it seems strange but I modified the value as your result.
Please check it.

**Review by masayuki2009 — APPROVED** [review]

**Pull request merged**